service/controller/resource/secretfinalizer/secret_accessor.go

```diff
@@ -20,7 +20,7 @@ func newSecretAccessors(ctx context.Context, cr infrastructurev1alpha2.AWSCluste
 		// certificate.
 		{
 			Name:      fmt.Sprintf("%s-api", key.ClusterID(&cr)),
-			Namespace: "default",
+			Namespace: cr.GetNamespace(),
```

Comment:
From a security perspective, will that be ok? The thing is it might be than possible for someone from the customer side to get access to the cert key as soon as they have access via RBAC to that namespace?

Reply:
That is a very good question.
Since right now everything is in the default namespace, wouldn't it at least make the access scope smaller?

Reply:
What would a customer need this for?
I don't see an immediate reason why a customer would need to access this secret. Presumably, there should be individual keypairs/kubeconfigs for each customer user which may have lower access than this, so I would be inclined to keep these outside customer-facing namespaces.

Reply:
That makes sense. I was approaching it from the perspective that all resources belonging to a specific cluster should move from the default to a configurable namespace. But this might not make sense for secrets when we create the namespaces by organization.

Reply:
For sure, I agree with separating them into dedicated namespaces. Since presumably the customer will get access to `default` as well, maybe it makes sense to have an additional namespace per-cluster for things like this which Giant Swarm needs in order to manage the cluster, but which the customer doesn't need access to.

Reply:
That sounds like a very good separation. We should have a separate story.
As for the configurable namespace for cluster resources I believe there is nothing more to change in `cluster-operator`